include/envoy/upstream/upstream.h

```diff
@@ -80,11 +81,19 @@ class Host : virtual public HostDescription {
   virtual bool healthy() const PURE;
 
   /**
-   * Set the host's outlier detector. Outlier detectors are assumed to be thread safe, however
-   * a new outlier detector must be installed before the host is used across threads. Thus,
+   * Set the host's health checker monitor. Monitors are assumed to be thread safe, however
+   * a new monitor must be installed before the host is used across threads. Thus,
    * this routine should only be called on the main thread before the host is used across threads.
```

Comment:
Maybe add an `ASSERT` verifying we're on the main thread in the implementation?

Reply:
There is no great way to do this because of how we use hosts in tests, etc. I think I'm going to skip this for now. I will make a note to look at it in a follow up.